core/src/functionalTest/groovy/com/novoda/gradle/release/AndroidDifferentGradleVersions.groovy

```diff
@@ -0,0 +1,134 @@
+
+    @Rule
+    public TemporaryFolder temporaryFolder = new TemporaryFolder()
+
+    private GradleRunner runner
+
+    @Before
+    void setUp() throws Exception {
+        runner = GradleRunner.create()
+                .withProjectDir(temporaryFolder.root)
+                .withArguments("build", "bintrayUpload", "-PbintrayKey=key", "-PbintrayUser=user")
+                .withPluginClasspath()
+
+        File manifestFile = new File(temporaryFolder.root, "/src/main/AndroidManifest.xml")
+        manifestFile.getParentFile().mkdirs()
+        manifestFile.createNewFile()
+        manifestFile.write('<manifest package="com.novoda.test"/>')
+
+        def buildFile = temporaryFolder.newFile("build.gradle")
+        buildFile.write(
+                """            
+            buildscript {
+                repositories {
+                    jcenter()
+                    google()
+                }
+                dependencies {
+                    classpath 'com.android.tools.build:gradle:3.0.0'
+                    classpath 'com.novoda:bintray-release:0.8.0'
+                }
+            }
+            
+            apply plugin: "com.android.library"
+            apply plugin: "com.novoda.bintray-release"
+            
+            android {
+                compileSdkVersion 26
+                buildToolsVersion "26.0.2"
+
+                defaultConfig {
+                    minSdkVersion 16
+                    versionCode 1
+                    versionName "0.0.1"
+                }    
+                
+                lintOptions {
+                   abortOnError false
+                }
+            }
+            
+            repositories {
+                jcenter()
+            }
+            
+            dependencies {
+                implementation "junit:junit:4.12"
+            }
+            
+            publish {
+                userOrg = 'novoda'
+                groupId = 'com.novoda'
+                artifactId = 'test'
+                publishVersion = '1.0'
+            }
+                    """
+        )
+
+        def fakeSourceCode = new File(temporaryFolder.root, "src/main/java/HelloWorld.java")
+        fakeSourceCode.getParentFile().mkdirs()
+        fakeSourceCode.createNewFile()
+        fakeSourceCode.write(" public class HelloWorld {} ")
+    }
+
+    @After
+    void tearDown() throws Exception {
+        temporaryFolder.delete()
+    }
+
+    /**
+     * Android don't support only gradle 4.1 and up with plugin 3.0
+     */
+    @Test
+    void test_withGradle40_shouldFail() {
+        runner.withGradleVersion("4.0").buildAndFail()
+    }
+
+    @Test
+    void test_withGradle41_shouldSucceed() {
+        def result = runner.withGradleVersion("4.1").build()
+
+        Assertions.assertThat(result.task(":bintrayUpload").outcome).isEqualTo(TaskOutcome.SUCCESS)
+    }
+
+    @Test
+    void test_withGradle42_shouldSucceed() {
+        def result = runner.withGradleVersion("4.2").build()
```

Comment:
What about a `Parameterized` test with 2 dimensional parameters. 
First param is the gradle version and second is the expected outcome.

So when a new version of gradle is released we can just modify params instead of copy pasting tests.

Reply:
I don't really get it 🤔 

You mean having one test which runs different gradle versions and checks the output? 🤔 
Don't know if I like it because the report would be that "the test is broken" and not "test gradle42 is broken"...

Reply:
You will write and maintain one test but actually they will be run separately. You can also name them.
Have a look at Parameterized test from JUnit. Also used in our static analysis plugin tests

Reply:
Ah ok, now I get it.
I didn't know the [Parameterized test](https://github.com/junit-team/junit4/wiki/parameterized-tests) from junit. Thanks for the hint 👍